src/core_plugins/kibana/public/management/sections/objects/change_index_modal.js

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { groupBy, mapValues, take } from 'lodash';
+
+import {
+  KuiModal,
+  KuiModalHeader,
+  KuiModalHeaderTitle,
+  KuiModalBody,
+  KuiModalBodyText,
+  KuiModalFooter,
+  KuiButton,
+  KuiModalOverlay,
+  KuiTable,
+  KuiTableBody,
+  KuiTableHeader,
+  KuiTableHeaderCell,
+  KuiTableRow,
+  KuiTableRowCell,
+  KuiControlledTable,
+  KuiToolBar,
+  KuiToolBarSection,
+  KuiPager,
+} from 'ui_framework/components';
+
+export class ChangeIndexModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const byId = groupBy(props.conflictedObjects, ({ obj }) => obj.searchSource.getOwn('index'));
+    this.state = {
+      page: 0,
+      perPage: 10,
+      objects: mapValues(byId, (list, indexPatternId) => {
+        return {
+          newIndexPatternId: props.indices[0].id,
+          list: list.map(({ doc }) => {
+            return {
+              id: indexPatternId,
+              type: doc._type,
+              name: doc._source.title,
+            };
+          })
+        };
+      })
+    };
+  }
+
+  changeIndex = () => {
+    const result = Object.keys(this.state.objects).map(indexPatternId => ({
+      oldId: indexPatternId,
+      newId: this.state.objects[indexPatternId].newIndexPatternId,
+    }));
+    this.props.onChange(result);
+  };
+
+  onIndexChange = (id, event) => {
+    event.persist();
+    this.setState(state => {
+      return {
+        ...state,
+        objects: {
+          ...state.objects,
+          [id]: {
+            ...state.objects[id],
+            newIndexPatternId: event.target.value,
+          }
+        }
+      };
+    });
+  };
+
+  onKeyDown = (event) => {
+    if (event.keyCode === 27) { // ESC key
```

Comment:
You can remove the comment here and instead import `keyCodes` from the UI Framework:

```javascript
import { keyCodes } from 'ui_framework/services';

if (event.keyCode === keyCodes.ESCAPE) {
```